### PR TITLE
Add OnResume, OnAVChange and OnAVStart to jsonRPC notifications

### PR DESCRIPTION
**Pull request by razzeee**
## Description
As we did for the python interface I added OnAVChange and OnAVStart and took the liberty to also add OnResume. Which will hopefully offer richer messaging and be easier to handle. Be aware that OnPlay will only be called at start now and not when resuming.
Not 100% sure about the version increase. (as always)

## Motivation and Context

Adds the new events to the jsonRPC and untangled OnPlay useage a bit.

## How Has This Been Tested?

Tested locally on windows 10. See screenshot.

## Screenshots (if appropriate):
grafik

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [X] My change requires a change to the documentation, either Doxygen or wiki
- [X] I have updated the documentation accordingly
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [X] All new and existing tests passed

### CODE REVIEW

**Comment by FernetMenta** [comment]
thank you very much!

File: xbmc/Application.cpp [hunk 1]
```diff
@@ -3439,7 +3439,7 @@ void CApplication::OnPlayBackResumed()
   CVariant param;
   param["player"]["speed"] = 1;
   param["player"]["playerid"] = CServiceBroker::GetPlaylistPlayer().GetCurrentPlaylist();
-  CAnnouncementManager::GetInstance().Announce(Player, "xbmc", "OnPlay", m_itemCurrentFile, param);
+  CAnnouncementManager::GetInstance().Announce(Player, "xbmc", "OnResume", m_itemCurrentFile, param);
```
Comment (afedchin) [inline xbmc/Application.cpp]: please add handling new event here please https://github.com/xbmc/xbmc/blob/master/xbmc/windowing/win10/WinEventsWin10.cpp#L618

**Review by afedchin — CHANGES_REQUESTED** [review]
breaks current functionality

**Comment by Razzeee** [comment]
## Learning
If you cmake for one platform Visual Studio will only search those files. So you will have to grep. I recommend ripgrep for that https://github.com/BurntSushi/ripgrep

## This got a whole lot more complicated 😄 
I will have to check if android is still fine with these changes as I decided to call the `onResume` method included in the main android file. Otherwise I only replicate old behavior.

File: xbmc/Application.cpp (hunk 1 above)
Comment (Razzeee) [reply xbmc/Application.cpp]: Done, thanks!

**Review by afedchin — APPROVED** [review]

**Comment by Razzeee** [comment]
I removed the before mentioned call to `onResume` in `xbmcApp.cpp` for android. As the `onResume` method wasn't for playback resuming, but for activity (as in android activity) resuming.

**Comment by Razzeee** [comment]
OSX build error seems unrelated

**Comment by DaveTBlake** [comment]
>Not 100% sure about the version increase. (as always)

v 9.2.0 is correct. This is a breaking change and strictly a major verion step (e.g. 10), but rather than have runnaway major version numbers @Montellese advises that the major version only changes with the first breaking change made during alpha, and subsequent ones only step the minor version. 

But it is also important to start the version.txt changing commit title with *[Breaking change]*, to make it easy to spot in the commit history. Remember Git labels don't appear in commit history, the clues need to be in the titles. Please change this.

API consumers like @SyncedSynapse and @joethefox may also like to know that this change is happening. If they are using onPlay announcements after his change they will need to start listening to onResume.

**Comment by Razzeee** [comment]
@DaveTBlake prefixed the commit msg.

I would like to merge this on friday, if nobody speaks up.

**Comment by Razzeee** [comment]
Wiki updated https://kodi.wiki/view/JSON-RPC_API/v9#Player_3

**Pull request merged**